src/decimal.rs

```diff
@@ -637,6 +637,102 @@ impl Decimal {
         Ok(ret)
     }
 
+    /// Returns a `Result` which if successful contains the `Decimal` constitution of
+    /// the scientific notation provided by `value`. If the exponent is negative and
+    /// the given base and exponent would exceed [Decimal::MAX_SCALE] then this
+    /// functions attempts to round the base to fit.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - The scientific notation of the `Decimal`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use rust_decimal::Decimal;
+    /// # use rust_decimal::Error;
+    /// #
+    /// # fn main() -> Result<(), rust_decimal::Error> {
+    /// let value = Decimal::from_scientific_round("2.710505431213761e-20")?;
+    /// assert_eq!(value.to_string(), "0.0000000000000000000271050543");
+    ///
+    /// let value = Decimal::from_scientific_round("2.5e-28")?;
+    /// assert_eq!(value.to_string(), "0.0000000000000000000000000003");
+    ///
+    /// let value = Decimal::from_scientific_round("-2.5e-28")?;
+    /// assert_eq!(value.to_string(), "-0.0000000000000000000000000003");
+    ///
+    /// let err = Decimal::from_scientific_round("2e-29").unwrap_err();
+    /// assert_eq!(err, Error::ScaleExceedsMaximumPrecision(29));
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub fn from_scientific_round(value: &str) -> Result<Decimal, Error> {
```

Comment:
I'm not a big fan of this name, but I get where you're coming from. What about `from_scientific_lossy`? I only pick that up since it loosely matches a Rust naming convention (e.g. `from_utf8_lossy`).